src/main/java/org/threadly/litesockets/networkutils/ProfileServer.java

```diff
@@ -0,0 +1,213 @@
+package org.threadly.litesockets.networkutils;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.threadly.concurrent.AbstractService;
+import org.threadly.concurrent.SimpleSchedulerInterface;
+import org.threadly.litesockets.Client;
+import org.threadly.litesockets.Client.Closer;
+import org.threadly.litesockets.Client.Reader;
+import org.threadly.litesockets.Server.ClientAcceptor;
+import org.threadly.litesockets.SocketExecuterInterface;
+import org.threadly.litesockets.tcp.TCPServer;
+import org.threadly.litesockets.utils.MergedByteBuffers;
+import org.threadly.util.ExceptionUtils;
+import org.threadly.util.debug.Profiler;
+
+
+/**
+ * The ProfileServer Uses Threadlys {@link Profiler} tying it to a listen socket to make it easy
+ * to profile running java processes as needed. 
+ *
+ * This will open a listen port on specified ip/port and allow connections to send basic commands to the
+ * process to control the profiler.
+ * 
+ * Here are a list of commands:
+ * 
+ * start - Starts the profiler
+ * stop - Stops the profiler (Profile is maintained)
+ * reset - Resets the profilers data
+ * dump - dumps the profiler current data
+ * 
+ * Commands must end with a newline.
+ * 
+ * NOTE: the profiler should only be used when needed and stopped and reset when not in use.  The longer it
+ * runs the more CPU and Memory it will consume, to the point where it could eat up an entire CPU core and over fill
+ * memory.
+ *
+ */
+public class ProfileServer extends AbstractService implements ClientAcceptor, Reader, Closer{
+  protected static final ByteBuffer DUMP_EXCEPTION = ByteBuffer.wrap("Got Exception doing Dump!\n\n".getBytes()).asReadOnlyBuffer();
```

Comment:
Its not really supposed to change based on OS.  Honestly I doubt it would mater I dont think telnet from windows does or atleast cares about \r